obelisk/python/obelisk_py/node.py

```diff
@@ -1,86 +1,515 @@
+            required_field_names: The required field names.
+            optional_field_names: The optional field names.
+
+        Raises:
+            AssertionError: If the configuration string is invalid.
+        """
+        assert all(
+            [field in field_names for field in required_field_names]
+        ), f"config_str must contain the following fields: {required_field_names}"
+        assert all([field in required_field_names + optional_field_names for field in field_names]), (
+            f"""The following fields in the config_str are invalid: {
+                set(field_names) - set(required_field_names + optional_field_names)
+            }""",
+            f"Currently-supported fields in Obelisk are: {required_field_names + optional_field_names}",
+        )
+
+    @staticmethod
+    def _check_values(value_names: List[str], allowable_value_names: List[str]) -> None:
+        """Check if the values in a configuration string are valid.
+
+        Parameters:
+            value_names: The list of values.
+            allowable_value_names: The allowable values.
+
+        Raises:
+            AssertionError: If the values are invalid.
+        """
+        assert all([value in allowable_value_names for value in value_names]), (
+            f"""The following values in the config_str are invalid: {
+                set(value_names) - set(allowable_value_names)
+            }. """,
+            f"Currently-supported values in Obelisk are: {allowable_value_names}",
+        )
+
+    def _get_msg_type_from_config_dict(self, config_dict: Dict) -> Type:
+        """Get the message type from a configuration dictionary."""
+        assert config_dict.get("msg_type") is not None, "No message type supplied!"
+        assert isinstance(config_dict["msg_type"], str), "The 'msg_type' field must be a string!"
+
+        if "non_obelisk" in config_dict:
+            assert isinstance(config_dict["non_obelisk"], str), "The 'non_obelisk' field must be a string!"
+            assert config_dict["non_obelisk"].lower() != "true", "non_obelisk=True but no message type supplied!"
+            msg_type = check_and_get_obelisk_msg_type(self, config_dict["msg_type"], ObeliskMsg)
+        else:
+            self.get_logger().warn(
+                "Creating a publisher that can publish non-Obelisk messages. "
+                "This may cause certain API incompatibilities."
+            )
+            msg_type = check_and_get_obelisk_msg_type(self, config_dict["msg_type"], ObeliskAllowedMsg)
+
+        return msg_type
+
+    def _get_callback_group_from_config_dict(self, config_dict: Dict) -> Optional[CallbackGroup]:
+        """Get the callback group from a configuration dictionary."""
+        if "callback_group" in config_dict:
+            assert isinstance(config_dict["callback_group"], str), "The 'callback_group' field must be a string!"
+            if config_dict["callback_group"].lower() == "none":
```

Comment:
Might want to make "none" -> "None" because iirc all the other fields are CamelCase.

Reply:
this is fine, the `.lower()` call forcibly lower cases everything to check string matching. I think it's actually ok if the user passes things like `none` as well.